### DISCUSSION TITLE
Google apps script upload image

### DISCUSSION THREAD

**Discussion by Topdev428** [community/community#118300] [Other Feature Feedback, Questions, & Ideas] [answered]
### Body

I'm gonna build the google apps script project.
but I can't carry out image upload in google driver.
Please help me

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by SrDev36428** [comment, marked as answer]
Very interesting.
I've already experienced in a project like your one.
So, I can show me the source.
This is Frontend part
const uploadFiles = () => {
    const fileInput = document.getElementById('virtualPhoto');
    const files = fileInput.files;

    if (!files || files.length === 0) {
      document.getElementById("photourl").value = "";
      alert('Please select one or more files.');
      return;
    }

    const imageFiles = [];
    for (let i = 0; i < files.length; i++) {
      const file = files[i];
      if (file.type.startsWith('image/')) { // Check if file is an image
        imageFiles.push(file);
      } else {
        document.getElementById("mask").style.display = "none";
        document.getElementById("photourl").value = "";
        alert('Please select only image files.');
        return;
      }
    }

    const filesData = [];
    for (let i = 0; i < imageFiles.length; i++) {
      const file = imageFiles[i];
      const fr = new FileReader();
      fr.onload = function (e) {
        const fileData = {
          filename: generateFilename(file.name, i),
          mimeType: file.type,
          bytes: [...new Int8Array(e.target.result)]
        };
        filesData.push(fileData);
        if (filesData.length === imageFiles.length) {
          upload(filesData);
        }
      };
      fr.readAsArrayBuffer(file);
    }
  }

  const upload = (filesData) => {
    google.script.run.withSuccessHandler(displayUrls).saveFilesToDrive(filesData);
  }

....


This is backend part.

function saveFilesToDrive(filesData, folderId) {
  try {
    var folder = DriveApp.getFolderById(photofolderID);
    var urls = []
    for (var i = 0; i < filesData.length; i++) {
      var fileData = filesData[i];
      var blob = Utilities.newBlob(fileData.bytes, fileData.mimeType, fileData.filename);
      var file = folder.createFile(blob);
      urls.push(file.getUrl())
    }
    return urls;
  } catch (error) {
    return error.toString();
  }
}

I hope this is helpful for you

**Reply by Topdev428** [reply]
perfect.
It works very well.
I appreciate your help.

**Reply by ThynkCraft** [reply]
Me too.
It is helpful for me.
Thank you

**Comment by Topdev428** [comment]
Devsworld36428 , you are very excellent

**Comment by SrDev36428** [comment]
Thank you very much
